packages/commons/src/auth/authorization.ts

```diff
@@ -0,0 +1,53 @@
+import { unauthorizedError } from "pagopa-interop-models";
+import { AppContext, AuthData, UserRole } from "../index.js";
+
+export function hasUserRole(
+  authData: AuthData,
+  userRoles: UserRole[]
+): boolean {
+  return (
+    authData.tokenType === "ui" &&
+    authData.userRoles.some((role: UserRole) => userRoles.includes(role))
+  );
+}
+
+export function validateAuthorization<
+  T extends Exclude<AuthData["tokenType"], "ui">
+>(
+  ctx: AppContext,
+  tokenTypes: T[]
+): asserts ctx is AppContext<Extract<AuthData, { tokenType: T }>>;
+export function validateAuthorization<T extends AuthData["tokenType"]>(
+  ctx: AppContext,
+  tokenTypes: T[],
+  admittedUserRoles: UserRole[]
+): asserts ctx is AppContext<Extract<AuthData, { tokenType: T }>>;
+export function validateAuthorization<T extends AuthData["tokenType"]>(
+  ctx: AppContext,
+  tokenTypes: T[],
+  admittedUserRoles?: UserRole[]
```

Comment:
We may enforce the minimum 1 on the array also in the type system

```suggestion
  tokenTypes: [T, ...T[]]
): asserts ctx is AppContext<Extract<AuthData, { tokenType: T }>>;
export function validateAuthorization<T extends AuthData["tokenType"]>(
  ctx: AppContext,
  tokenTypes: [T, ...T[]],
  admittedUserRoles: [UserRole, ...UserRole[]]
): asserts ctx is AppContext<Extract<AuthData, { tokenType: T }>>;
export function validateAuthorization<T extends AuthData["tokenType"]>(
  ctx: AppContext,
  tokenTypes: T[],
  admittedUserRoles?: [UserRole, ...UserRole[]]
```

Reply:
Done!